halyard-web/src/main/java/com/netflix/spinnaker/halyard/model/v1/providers/Provider.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright 2016 Google, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.halyard.model.v1.providers;
+
+import java.util.List;
+
+public class Provider<T extends Account> {
+  public boolean enabled;
```

Comment:
I'm going to augment your question to be "what's the detriment to public _fields within data classes_"

A few reasons that by themselves aren't convincing, but altogether seem compelling to me:

1. External classes coupling directly to a field mean you don't know who needs access. Put another way, you can't set a debug breakpoint on a getter and track back to where the data is giong.
2. External classes can change the value. [This post](http://stackoverflow.com/questions/7959129/whats-the-deal-with-javas-public-fields/7959871#7959871) advocates making public fields final, which I think isn't a bad option.
3. The inability to modify the value on the fly (via overridden getter) should your future needs change.
4. If you ever decide to build up a hierarchy interfaces for related objects, your interfaces can only specify getters/setters. If that happens, you now have two ways to access your field - directly and through the getter, and that will cause confusion later on.